### PR TITLE
ensure check of offset repo is wrapped with 'eventually'

### PR DESCRIPTION
**Pull request by octonato**
References #643

### CODE REVIEW

File: akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala [hunk 1]
```diff
@@ -206,6 +206,13 @@ class CassandraProjectionSpec
 
   private def concatHandlerFail4(): ConcatHandlerFail4 = new ConcatHandlerFail4
 
+  def offsetShouldBe(projectionId: ProjectionId, expectedOffset: Long) = {
+    eventually {
+      val currentOffset = offsetStore.readOffset[Long](projectionId).futureValue
+      currentOffset shouldBe Some(expectedOffset)
+    }
+  }
```
Comment (octonato) [inline akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala:214]: cassandra use atLeastOnce so we should always use `eventually` when checking the offset.

File: akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala [hunk 2]
```diff
@@ -254,18 +258,14 @@ class CassandraProjectionSpec
...
-          val concatStr = repository.findById(entityId).futureValue
-          concatStr should matchPattern {
-            case Some(ConcatStr(_, "abc|def|ghi")) =>
-          }
+
+      offsetShouldBe(projectionId, 3L)
+      withClue("check: projection is consumed up to third") {
+        val concatStr = repository.findById(entityId).futureValue
+        concatStr should matchPattern {
+          case Some(ConcatStr(_, "abc|def|ghi")) =>
```
Comment (octonato) [inline akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala:266]: in a few places, I swapped the checks. 

If the offset pass the check, then it's guaranteed that the concat text is also updated and we don't need another eventually around the `concatStr` checks.

**Review by octonato — COMMENTED** [review]
There is one test I didn't touch because already fixed in #646 and the code changed a lot for this test. 

Hopefully the build won't fail on that un-refactored test.

**Comment by octonato** [comment]
failed because of JDBC, coincidentally a similar test

```scala
[info] - atLeastOnceShouldRestartFromPreviousOffset *** FAILED ***
[info]   org.junit.ComparisonFailure: expected:<abc|def|[ghi|]> but was:<abc|def|[]>
[info]   at org.junit.Assert.assertEquals(Assert.java:117)
[info]   at org.junit.Assert.assertEquals(Assert.java:146)
[info]   at akka.projection.jdbc.JdbcProjectionTest.lambda$atLeastOnceShouldRestartFromPreviousOffset$2e5cadc3$1(JdbcProjectionTest.java:342)
[info]   at akka.projection.testkit.javadsl.ProjectionTestKit.$anonfun$runWithTestSink$1(ProjectionTestKit.scala:98)
[info]   at akka.projection.testkit.javadsl.ProjectionTestKit.$anonfun$runWithTestSink$1$adapted(ProjectionTestKit.scala:98)
[info]   at akka.projection.testkit.scaladsl.ProjectionTestKit.runWithTestSink(ProjectionTestKit.scala:126)
[info]   at akka.projection.testkit.javadsl.ProjectionTestKit.runWithTestSink(ProjectionTestKit.scala:98)
[info]   at akka.projection.jdbc.JdbcProjectionTest.atLeastOnceShouldRestartFromPreviousOffset(JdbcProjectionTest.java:331)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[info]   at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
```

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**